src/LanguageServer/Impl/Diagnostics/DiagnosticsService.cs

```diff
@@ -64,7 +71,12 @@ private sealed class DocumentDiagnostics {
         private IRunningDocumentTable _rdt;
         private DateTime _lastChangeTime;
 
-        private IRunningDocumentTable Rdt => _rdt ?? (_rdt = _services.GetService<IRunningDocumentTable>());
+        private IRunningDocumentTable Rdt {
+            get {
+                ConnectToRdt();
```

Comment:
call ConntectToRdt to make sure event are always get registered. and _rdt are set under lock.